### PR TITLE
K8s: Add file store tests

### PR DESCRIPTION
**Pull request by toddtreece**
**What is this feature?**

[Add a brief description of what the feature or update does.]

**Why do we need this feature?**

This includes partial support for k8s storage tests. Work on additional test coverage will happen in a follow up PR.

**Who is this feature for?**

[Add information on what kind of user the feature is for.]

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: pkg/apiserver/storage/file/file.go [hunk 1]
```diff
@@ -482,35 +458,46 @@ func (s *Storage) GetList(ctx context.Context, key string, opts storage.ListOpti
...
+			expectedRV, err := s.versioner.ParseResourceVersion(opts.ResourceVersion)
+			if err != nil {
+				return err
+			}
+			if currentVersion != expectedRV {
 				continue
 			}
 		}
 
+		//if opts.SendInitialEvents == nil || (opts.SendInitialEvents != nil && !*opts.SendInitialEvents) {
```
Comment (ryantxu) [inline pkg/apiserver/storage/file/file.go:481]: ??? comment out or remove?

File: pkg/apiserver/storage/file/file.go [hunk 2]
```diff
@@ -15,9 +15,9 @@ import (
 	"sync"
 	"time"
 
-	"github.com/bwmarrin/snowflake"
 	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/api/meta"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
```
Comment (ryantxu) [inline pkg/apiserver/storage/file/file.go:20]:
```suggestion
	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
```
we use this in most other places

Comment (toddtreece) [reply pkg/apiserver/storage/file/file.go]: updated in https://github.com/grafana/grafana/pull/90151/commits/7d173efe465bffe194460deea2ae65a75c142126

File: pkg/apiserver/storage/file/file.go (hunk 1 above)
Comment (toddtreece) [reply pkg/apiserver/storage/file/file.go]: removed in https://github.com/grafana/grafana/pull/90151/commits/7d173efe465bffe194460deea2ae65a75c142126

**Review by ryantxu — APPROVED** [review]
LGTM -- I vote we comment out/skip the not passing ones and then merge.

These tests are pretty finicky and seem to require/assume an etcd backend.  Merging now will let base the new backend interface off this one more easily

**Pull request merged**